### ISSUE TITLE
unable to bundle numpy

### ISSUE THREAD

**Issue by tku137** [pyinstaller/pyinstaller#1886]
My system: Win10, Anaconda3

Small PyQt4 app bundles perfectly into single exe file. But when simply putting `import numpy` anywhere, PyInstaller bundles a bunch of definitely unused modules and the exe crashes with `pyi_rth_pkgres returned -1`.

What I tried:
- latest development version [d00163f](https://github.com/pyinstaller/pyinstaller/commit/d00163fed378d499ae0d709255da739145463c1b) using `pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip` from [How to Report Bugs](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs) results in broken exe with or without importing numpy, app gives `pyiboot01_bootstrap returned -1`
- `--onedir` and `--onefile`
- `--hidden-import numpy`
- tried to `--exclude-module` the unnecessarily imported modules, no effect
- specify `--additional-hooks-dir` with the `hook-numpy.py` from #1881 
- try with installed vanilla numpy without mkl bindings
- downgraded setuptools to 19.2 and some random other versions, can't really remember where I read that this might be a solution

I cleared the cache `%APPDATA%\Roaming\pyinstaller` prior to every try.

**Comment by htgoebel** [comment]
Thanks for the list of what you already tried.
* When using the develop version on Windows, you need to compile the bootloader since we did not yet update it since the last changes.
* Did you try release 3.1.1?
* Please provide the last part of the traceback. `pyi_rth_pkgres returned -1` is not quite informative.

**Comment by tku137** [comment]
I tried to build the bootloader, but I am not quite sure if I have done everything correctly. The docs are a bit confusing. Here is what I did according to the steps described [here](https://github.com/pyinstaller/pyinstaller/wiki/Build-Bootloader-Win64) and [here](https://pythonhosted.org/PyInstaller/#building-the-bootloader):

- already had VS installed, no need to do
- installed [the exact py27 version mentioned here](https://github.com/pyinstaller/pyinstaller/wiki/Build-Bootloader-Win64#python-27)
- run all steps exactly as described [here](https://github.com/pyinstaller/pyinstaller/wiki/Build-Bootloader-Win64#building)
- steps mentioned in [testing the build](https://github.com/pyinstaller/pyinstaller/wiki/Build-Bootloader-Win64#testing-the-build) were not possible, according to the readme.txt in the tests folder, one has to go in the subfolder `functional`, I run all tests, no response so I guess it's ok
- bundling did not work at this point, which is logical because the steps described in [building the bootloader](https://github.com/pyinstaller/pyinstaller/wiki/Build-Bootloader-Win64) seem to just create the source
- in the [PyInstaller Manual](https://pythonhosted.org/PyInstaller/#building-the-bootloader) it is mentioned to do some more steps, which I did, but I am not quite sure where to go from here. 

In the site-packages folder of the pyinstaller installation, there are some `run*.exe` files (and only them) under the `bootloader` folder, so I guessed I need to copy the `run*.exe` files from the `build` subfolder from the git-cloned folder, in which the previous steps took place, to the site-packages folder of my pyinstaller site-packages folder. However, this did not change anything.

I also tried to run `python setup.py install` from the git-cloned and built folder from above steps. Bundling still doesn't work.

Regarding the second question, yes I used the standard 3.1.1 release all the time. The first paragraph before the list of things I tried describes the error happening with this version. The things I tried were actually done with both release and development version.

Regarding the third qeustion: I bundled my app with the last stand after doing all the steps mentioned above in this comment, but without the `--windowed` flag, I hope this is what you need to know: console output is the following:

> Failed to execute script pyi_rth_pkgres
Traceback (most recent call last):
  File "PyInstaller\loader\rthooks\pyi_rth_pkgres.py", line 11, in <module>
  File "C:\Anaconda3\lib\site-packages\pyinstaller-3.2.dev0+gd00163f.mod-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_module
    exec(bytecode, module.__dict__)
  File "pkg_resources\__init__.py", line 75, in <module>
  File "pkg_resources\extern\__init__.py", line 60, in load_module
ImportError: The 'packaging' package is required; normally this is bundled with this package so if you get this warning, consult the packager of your distribution.

**Comment by LevN0** [comment]
Looks like a duplicate of this bug. Try downgrading setuptools to 19.2?

**Comment by tku137** [comment]
Ok, after a lot of research and testing, I finally made it work.

Downgrading setuptools as in #1781 did the trick for the previous errors and steps made. After that another error occured, see #1865 for solution. Then I needed to implement #1881. After that, scikit-image throw new errors, which were solvable by using a hidden import to `skimage._shared.transform`. Unfortunately, the `skimage.io.imread` module wouldn't let itself bundle due to missing I/O plugins, even trying solutions from [here](https://github.com/pyinstaller/pyinstaller/issues/583) and [here](http://stackoverflow.com/questions/34761862/pyinstaller-you-may-load-i-o-plugins-with-the-skimage-io-use-plugin), so I got rid of it and used `scipy.ndimage.imread` instead.

After all these steps, pyinstaller finally bundles a functional single-file exe. THANKS @htgoebel and @LevN0, I was about to give up on this :)
